### PR TITLE
feat: Add input type - RotatePanInput

### PR DESCRIPTION
**Pull request by jongmoon**
## Issue
#126

### CODE REVIEW

File: src/inputType/RotatePanInput.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		}
+
+		this.lastDiff = diff;
+		this.observer.change(this, event, toAxis(this.axes, [-diff])); // minus for clockwise
+	}
+
+	private triggerAnimation(event) {
+		const vx = event.velocityX;
+		const vy = event.velocityY;
+		const normalSpeed = Math.sqrt(vx * vx + vy * vy) * (this.lastDiff > 0 ? -1 : 1); // clockwise
```
Comment (WoodNeck) [inline src/inputType/RotatePanInput.ts]: Why does this called `normalSpeed`?
Does it means `normal` or `normalized`?
I think it should be called just `velocity` or `tangent` or something as it is neither on normal direction nor normalized.

File: src/inputType/RotatePanInput.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	private prevQuadrant: number;
+	private lastDiff: number;
+	private coefficientForDistanceToAngle: number;
+
+	constructor(el: string | HTMLElement, options?: PanInputOption) {
+		super(el, options);
+
+		const rect = this.element.getBoundingClientRect();
+		// TODO: how to do if element is ellipse not circle.
+		this.coefficientForDistanceToAngle = 360 / (rect.width * Math.PI); // from 2*pi*r * x / 360
```
Comment (WoodNeck) [inline src/inputType/RotatePanInput.ts]: Should it have to be different with circle and ellipse?
I think we can generate degree by center position, and calculating position.

File: src/inputType/RotatePanInput.ts (hunk 1 above)
Comment (jongmoon) [reply src/inputType/RotatePanInput.ts]: @WoodNeck I had just taken variable names used in `PanInput.ts` But I think it's right to have meaning full name.

File: src/inputType/RotatePanInput.ts (hunk 2 above)
Comment (jongmoon) [reply src/inputType/RotatePanInput.ts]: @WoodNeck 
`coefficentForDistanceToAngle` is for calculating coefficient to convert from circumference(by user input distance) to angle.

But the circumference would be different between circle and ellipse.  So I had commented it (for later support)

File: src/inputType/RotatePanInput.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		// TODO: provide a way to set origin like https://developer.mozilla.org/en-US/docs/Web/CSS/transform-origin
+		this.rotateOrigin = [rect.left + (rect.width - 1) / 2, rect.top + (rect.height - 1) / 2];
```
Comment (daybrush) [inline src/inputType/RotatePanInput.ts]: @jongmoon 
Did rotatorigin respond to resize?

Shouldn't it be added at the time of the hold event as well as the constructor?

File: src/inputType/RotatePanInput.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+	onPanend(event) {
+		this.triggerChange(event);
+		this.triggerAnimation(event);
+	}
+
+	private triggerChange(event) {
+		const angle = this.getAngle(event.center.x, event.center.y);
+		const quadrant = this.getQuadrant(event.center.x, event.center.y);
+		const diff = this.getDifference(this.prevAngle, angle, this.prevQuadrant, quadrant);
```
Comment (daybrush) [inline src/inputType/RotatePanInput.ts]: @jongmoon

Why don't we combine getAngle, getQuadrant, and getDifferences into one method?

File: src/inputType/RotatePanInput.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+		const duration = Math.abs(velocity / -this.observer.options.deceleration);
+		const distance = velocity / 2 * duration;
+
+		this.observer.release(this, event, toAxis(this.axes, [distance * this.coefficientForDistanceToAngle]));
+	}
+
+	private getDifference(prevAngle: number, angle: number, prevQuadrant: number, quadrant: number) {
+		let diff: number;
+
+		if (prevAngle === null || prevQuadrant === null) {
```
Comment (daybrush) [inline src/inputType/RotatePanInput.ts]: @jongmoon

Could quadrant be null?
I only see numbers when I look at the calculations.

File: src/inputType/RotatePanInput.ts (hunk 3 above)
Comment (jongmoon) [reply src/inputType/RotatePanInput.ts]: @daybrush I didn't consider 'responsive` but your suggestion would be better.

File: src/inputType/RotatePanInput.ts (hunk 4 above)
Comment (jongmoon) [reply src/inputType/RotatePanInput.ts]: I make method only if that is called twice. But they are called once.

File: src/inputType/RotatePanInput.ts (hunk 5 above)
Comment (jongmoon) [reply src/inputType/RotatePanInput.ts]: @daybrush I think you are right. It's unreachable condition.

**Pull request merged**